tests/src/Loader/binding/tracing/BinderTracingTest.cs

```diff
@@ -3,64 +3,332 @@
+                ResultAssemblyPath = asm.Location,
+                Cached = false
+            };
+        }
 
-                BindOperation[] binds = listener.WaitAndGetEventsForAssembly(assemblyName);
-                Assert.True(binds.Length == 1, $"Bind count for {assemblyName} - expected: 1, actual: {binds.Length}");
-                BindOperation bind = binds[0];
-                Assert.True(bind.Success, $"Expected bind for {assemblyName} to succeed");
+        [BinderTest]
+        public static BindOperation LoadFromAssemblyPath()
+        {
+            CustomALC alc = new CustomALC(nameof(LoadFromAssemblyPath));
+            var executingAssembly = Assembly.GetExecutingAssembly();
+            Assembly asm = alc.LoadFromAssemblyPath(executingAssembly.Location);
+
+            return new BindOperation()
+            {
+                AssemblyName = executingAssembly.GetName(),
+                AssemblyPath = executingAssembly.Location,
+                AssemblyLoadContext = alc.ToString(),
+                Success = true,
+                ResultAssemblyName = asm.GetName(),
+                ResultAssemblyPath = asm.Location,
+                Cached = false
+            };
+        }
+
+        [BinderTest(isolate: true)]
+        public static BindOperation LoadFrom()
+        {
+            var executingAssembly = Assembly.GetExecutingAssembly();
+            Assembly asm = Assembly.LoadFrom(executingAssembly.Location);
+
+            return new BindOperation()
+            {
+                AssemblyName = executingAssembly.GetName(),
+                AssemblyPath = executingAssembly.Location,
+                AssemblyLoadContext = DefaultALC,
+                Success = true,
+                ResultAssemblyName = asm.GetName(),
+                ResultAssemblyPath = asm.Location,
+                Cached = false
+            };
+        }
+
+        [BinderTest(isolate: true)]
+        public static BindOperation PlatformAssembly()
+        {
+            string assemblyName = "System.Xml";
+            Assembly asm = Assembly.Load(assemblyName);
+            
+            return new BindOperation()
+            {
+                AssemblyName = new AssemblyName(assemblyName),
+                AssemblyLoadContext = DefaultALC,
+                Success = true,
+                ResultAssemblyName = asm.GetName(),
+                ResultAssemblyPath = asm.Location,
+                Cached = false
+            };
+        }
+
+        [BinderTest]
+        public static BindOperation NonExistentAssembly()
+        {
+            string assemblyName = "DoesNotExist";
+            try
+            {
+                Assembly.Load(assemblyName);
             }
+            catch { }
+
+            return new BindOperation()
+            {
+                AssemblyName = new AssemblyName(assemblyName),
+                AssemblyLoadContext = DefaultALC,
+                Success = false,
+                Cached = false
```

Comment:
It seems we don't have a test case for operation which would return `Cached=true` - is that right?

Reply:
Wow, how did I miss that.